lib/rmt/mirror.rb

```diff
@@ -128,10 +127,10 @@ def mirror_license
         @downloader.download(filename)
       end
     rescue RMT::Downloader::Exception => e
-      FileUtils.remove_entry(@temp_licenses_dir)
-      @temp_licenses_dir = nil
       raise RMT::Mirror::Exception.new("Error during mirroring metadata: #{e.message}")
     end
+
+    replace_directory(@temp_licenses_dir, File.join(@repository_dir, '../product.license/'))
```

Comment:
Previously the replacing happened at the very end -- to make sure that you have consistency between metadata and the repo contents.
With this change, when an error occurs during mirroring of RPMs you'll end up with old repo metadata, old RPM and the new product license (which might not be too terrible by itself, but this behavior is inconsistent).

Reply:
I figured it was the easiest way to solve the issue, but you're right that it might create inconsistencies.

My other solution was to do something like

```ruby
def mirror
  create_directories
  license_mirrored = mirror_license # mirror_license returns true or false depending on success

  # (...)
  replace_directory(@temp_licenses_dir, File.join(@repository_dir, '../product.license/')) if license_mirrored
  replace_directory(File.join(@temp_metadata_dir, 'repodata'), File.join(@repository_dir, 'repodata'))
ensure
  remove_tmp_directories
end
```

What do you think?

Reply:
To address the issue at hand, which is the empty `product.license` directory, I would suggest to make changes here to remove the tempdir: https://github.com/SUSE/rmt/blob/9ba286fb871014db934f67559a7160e952166223/lib/rmt/mirror.rb#L118-L121

And then to skip `replace_directory` if there is no temp directory for licenses.

But refactoring to consolidate all of the temp directory cleanups under `mirror` method is a good idea.

Reply:
Yup that would also work.

Reply:
Updated.
I think my changes also fix #118.